src/main/scala/loamstream/googlecloud/DataProcClient.scala

```diff
@@ -4,10 +4,20 @@ package loamstream.googlecloud
  * @author clint
  * Nov 28, 2016
  */
-trait DataProcClient {
-  def startCluster(): Unit
+trait DataProcClient extends DataProcClient.CanStart with DataProcClient.CanStop with DataProcClient.CanTellIfRunning
+
+object DataProcClient {
```

Comment:
The traits defined in here seem like they'd be very useful outside of the `googlecloud` package and instead as part of some generic `cloud.google` package (thinking ahead to when you add AWS support). They are pretty generic.

Reply:
Agreed.  This was one of those cases where designing for testability led to decent abstractions.  We never planned to run LS at Amazon, say, but we didn't want to rule that out.  Mostly, we needed to describe operations that could happen via shelling out to CLI tools, maybe in the future calling some Java API, or in some mock way for tests.  That led to a high-level, just-the-facts API here at least.